minefield_game/statePattern/ResolutionState.cpp

```diff
@@ -0,0 +1,159 @@
+#include "GameContext.h"
+#include "ResolutionState.h"
+#include "EndGameState.h"
+#include "PlacementState.h"
+#include "../cell_header.h"
+#include <iostream>
+#include <algorithm>
+
+bool cellMatches(const Cell &a, const Cell &b)
+{
+    return a.getX() == b.getX() && a.getY() == b.getY();
+}
+
+void processGuesses(std::string const &attackerName, Player &attacker, Player &defender, Board &board)
+{
+    std::vector<Mine> updatedMines;
+    for (const auto &mine : defender.mines)
+    {
+        bool hit = false;
+        for (const auto &guess : attacker.guesses)
+        {
+            if (cellMatches(mine.cell, guess))
+            {
+                std::cout << attackerName << " hit a mine at (" << mine.cell.getX() << ", " << mine.cell.getY() << ")!\n";
+                hit = true;
+                break;
+            }
+        }
+        if (!hit)
+        {
+            updatedMines.push_back(mine);
+        }
+    }
+
+    // save destroyed mines
+    for (const auto &oldMine : defender.mines)
+    {
+        bool wasDestroyed = std::none_of(updatedMines.begin(), updatedMines.end(), [&](const Mine &m)
+                                         { return cellMatches(m.cell, oldMine.cell); });
```

Comment:
#nice usage of `<algorithm>` and lambdas!

But... #should not capture everything (that's what `[&]` is doing), just capture what you need instead (`[&oldMine](const Mine& m) { return cellMatches(m.cell, oldMine.cell); }`)